internal/generate/types.go

```diff
@@ -171,18 +261,18 @@ func constructParamTypes(paths map[string]*openapi3.PathItem) []TypeTemplate {
 					}
 
 					paramName := strcase.ToCamel(p.Value.Name)
+					paramType := convertToValidGoType("", "", p.Value.Schema)
 					field := TypeField{
-						Name: paramName,
-						Type: convertToValidGoType("", "", p.Value.Schema),
+						Name:       paramName,
+						Type:       paramType,
+						MarshalKey: p.Value.Name,
+						Schema:     nil, // nil so StructTag always uses omitempty
```

Comment:
Would setting `OmitDirective: "omitempty"` work as well here? instead of `Schema: nil`? It seems a bit more clear what the intention is without relying on the comment.